### PR TITLE
implement NonCallbackInterfaceObjectClass::as_jsclass()

### LINKED ISSUE #9287: Implement NonCallbackInterfaceObjectClass::as_class()
The goal here is to encapsulate the `&*(class as *const _ as *const JSClass)` in `create_noncallback_interface_object` into its own method.

Code: `components/script/dom/bindings/interface.rs`

### PR DESCRIPTION
**Pull request by apopiak**
close #9287

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9296)

### CODE REVIEW

**Comment by ecoal95** [comment]
Looks good to me once the nits are addressed :)

-S-awaiting-review +S-needs-code-changes 

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[components/script/dom/bindings/interface.rs, line 110 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9296#-K7wNPhs5j9y2UEV8_gV:-K7wNPhs5j9y2UEV8_gW:1653498270) ([raw file](https://github.com/servo/servo/blob/67ae1b6d5251079e08520ade1f73f78518500fc6/components/script/dom/bindings/interface.rs#L110)):*
nit: Remove white-space after the `&` character.

---

*[components/script/dom/bindings/interface.rs, line 156 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9296#-K7wNMJamOJeC7KRFotR:-K7wNMJamOJeC7KRFotS:-1637632856) ([raw file](https://github.com/servo/servo/blob/67ae1b6d5251079e08520ade1f73f78518500fc6/components/script/dom/bindings/interface.rs#L156)):*
Please remove this comment

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9296#-:-K7wNbivVHznQ9X-iJoV:-1820922495)*

**Comment by nox** [comment]
r-, as_class shouldn't be an unsafe method. It *is* safe to do it.

File: components/script/dom/bindings/interface.rs [hunk 1]
```diff
@@ -105,6 +105,11 @@ impl NonCallbackInterfaceObjectClass {
             },
         }
     }
+
+    /// cast own reference to `JSClass` reference
+    pub unsafe fn as_jsclass(& self) -> & JSClass {
+        &*(self as *const _ as *const JSClass)
```
Comment (KiChjang) [inline components/script/dom/bindings/interface.rs]: According to @nox's comment, he means that you should delete the `unsafe` keyword before the `fn` keyword, and put the function body inside an `unsafe` block.

Comment (apopiak) [inline components/script/dom/bindings/interface.rs]: will do

**Pull request merged**